specification/cosmos-db/resource-manager/Microsoft.DocumentDB/stable/2015-04-08/cosmos-db.json

```diff
@@ -1857,6 +2366,302 @@
         "modelAsString": false
       }
     },
+    "SqlDatabaseCreateUpdateParameters": {
+      "description": "Parameters to create and update Cosmos DB SQL database.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties to create and update Azure Cosmos DB SQL database.",
+          "$ref": "#/definitions/SqlDatabaseCreateUpdateProperties"
+        }
+      },
+      "required": [
+        "properties"
+      ]
+    },
+    "SqlDatabaseCreateUpdateProperties": {
+      "description": "Properties to create and update Azure Cosmos DB SQL database.",
+      "type": "object",
+      "properties": {
+        "resource": {
+          "description": "The standard JSON format of a SQL database",
+          "$ref": "#/definitions/SqlDatabaseResource"
+        },
+        "options": {
+          "description": "A key-value pair of options to be applied for the request. This corresponds to the headers sent with the request.",
+          "$ref": "#/definitions/CreateUpdateOptions"
+        }
+      },
+      "required": [
+        "resource",
+        "options"
+      ]
+    },
+    "SqlContainerCreateUpdateParameters": {
+      "description": "Parameters to create and update Cosmos DB SQL container.",
+      "type": "object",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties to create and update Azure Cosmos DB SQL container.",
+          "$ref": "#/definitions/SqlContainerCreateUpdateProperties"
+        }
+      },
+      "required": [
+        "properties"
+      ]
+    },
+    "SqlContainerCreateUpdateProperties": {
+      "description": "Properties to create and update Azure Cosmos DB SQL container.",
+      "type": "object",
+      "properties": {
+        "resource": {
+          "description": "The standard JSON format of a SQL container",
+          "$ref": "#/definitions/SqlContainerResource"
+        },
+        "options": {
+          "description": "A key-value pair of options to be applied for the request. This corresponds to the headers sent with the request.",
+          "$ref": "#/definitions/CreateUpdateOptions"
+        }
+      },
+      "required": [
+        "resource",
+        "options"
+      ]
+    },
+    "SqlDatabaseResource": {
+      "type": "object",
+      "description": "Cosmos DB SQL database id object",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Name of the Cosmos DB SQL database"
+        }
+      },
+      "required": [
+        "id"
+      ]
+    },
+    "SqlContainerResource": {
+      "type": "object",
+      "description": "Cosmos DB SQL container resource object",
+      "properties": {
+        "id": {
```

Comment:
I assume this is != the "id" in the top-level resource model (i.e. the ARM resource id). It would be preferred to use a different property name for this property under the "properties":{} object. For example, containerId or containerName.

Reply:
This is resource id

Reply:
It is not necessary to duplicate "id" underneath "properties"